content/blog/page-view-counter/index.md

```diff
@@ -0,0 +1,273 @@
+```csharp
+  public class ViewCount : TableEntity
+    {
+        public ViewCount(string URL)
+        {
+            this.PartitionKey = URL; //the partition key for load-balancing and identification
+            this.RowKey = "visits"; // hardcoded for identification
+            Count = 0; // the value we want to store
+        }
+
+        public int Count { get; set; }
+
+        public ViewCount()
+        {
+            Count = 0;
+        }
+    }
+```
+### Changing our function to connect to Storage
+
+We now have the entity that we want to store and retrieve, so we need to connect to Storage and do precisely that. Table entities are stored in Tables which are identified with a name, so we can choose it and connect to it through a `CloudTable` object. In general, the logic is as follows:
+
+1. Connect to the Storage Account through the connection string.
+1. Create a table client through which we do transactions (inserts, updates, etc.).
+1. Get the table through its name. 
+1. Execute transactions on the table.
+
+Here are the first 3 steps in code:
+
+```csharp
+    const string tableName = "viewcountertable";
+    
+    var storageAccount = CloudStorageAccount.Parse($"{yourStorageAccountConnectionString}");
+    var tableClient = storageAccount.CreateCloudTableClient();
+    CloudTable table = tableClient.GetTableReference(tableName);
+    
+    await table.CreateIfNotExistsAsync(); // we can let our code create the table if needed
+```
+
+We can now do transactions to the table! Let's get the URL parameter from the body first:
+
+```csharp
+    string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
+    dynamic data = JsonConvert.DeserializeObject(requestBody);
+    string pageViewURL = data?.URL;
+```
+
+And now, we can use `URL` to create a `ViewCount` object that uses it as a key. The logic should be easy to follow but here's a high-level overview:
+
+1. Try to retrieve the `ViewCount` object for the URL.
+1. If it doesn't exist, create a new `ViewCount` object (with count 0 by default) for that URL.
+1. Increase the view count on that object by 1.
+1. Store the updated entity.
+1. Return the view count to the function caller.
+
+Here's what the code looks like:
+
+```csharp
+     var retrievedResult = table.Execute(TableOperation.Retrieve<ViewCount>(pageViewURL, "visits")); // try to retrieve count for the URL
+     var pageViewCount = (ViewCount) retrievedResult.Result; // wait for the operation and get the result into a ViewCount object 
+  
+     pageViewCount = pageViewCount ?? new ViewCount(pageViewURL); // if the entity didn't exist, create a new one with count 0
+     pageViewCount.Count++; // augment the view count by one
+     
+     table.Execute(TableOperation.InsertOrReplace(pageViewCount)); // insert or replace the entity with the new count
+     return (ActionResult)new OkObjectResult(pageViewCount.Count.ToString()); // return the view count 
```

Comment:
comments on top for shorter lines for mobile (and probably for desktop too since these lines are very long)